lib/instrumentation/core/http2.js

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2025 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const NAMES = require('../../metrics/names')
+const recordExternal = require('../../metrics/recorders/http_external')
+const logger = require('../../logger').child({ component: 'http2' })
+const synthetics = require('../../synthetics')
+const urltils = require('../../util/urltils.js')
+const cat = require('#agentlib/util/cat.js')
+module.exports = initialize
+
+function initialize(agent, http2, moduleName, shim) {
+  shim.wrap(http2, 'connect', wrapConnect)
+  shim.wrapReturn(http2, 'connect', wrapSession)
```

Comment:
there's no reason to wrap and wrapReturn, you can just wrap, do what you need before, call the original, apply the instrumentation on result and return accordingly